packages/strapi-design-system/src/DatePicker/DatePickerCalendar.tsx

```diff
@@ -54,98 +76,328 @@ export const DatePickerCalendarContent = ({
   onChange,
   minDate,
   maxDate,
+  monthSelectLabel = 'Month',
+  yearSelectLabel = 'Year',
 }: DatePickerCalendarContentProps) => {
-  const [date, setDate] = useState(initialDate);
+  const [localDate, setLocalDate] = React.useState(() => convertUTCDateToCalendarDate(initialDate));
+  const startDate = startOfMonth(localDate);
+
+  const locale = defaultLocale || getDefaultLocale();
+
+  const years: string[] = React.useMemo(() => {
+    const minYear = minDate ? minDate.getFullYear() : initialDate.getFullYear() - DEFAULT_PAST_RANGE;
+    const maxYear = maxDate ? maxDate.getFullYear() : initialDate.getFullYear() + DEFAULT_FUTURE_RANGE;
+
+    return [...Array(maxYear - minYear + 1).keys()].map((y) => (minYear + y).toString());
+  }, [minDate, initialDate, maxDate]);
+
+  const monthFormatter = useDateFormatter(locale, { month: 'long' });
+  const months: string[] = React.useMemo(
+    () => [...Array(12).keys()].map((m) => monthFormatter.format(new Date(Date.UTC(2023, m)))),
```

Comment:
Can we derive how many month a year has from Intl? I know there are some with 10 and 13 months.

Reply:
I think the amount of months is derived from the calendar not the year but yeah we can do that.